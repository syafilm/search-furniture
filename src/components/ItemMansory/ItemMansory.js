import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Styles,
  Helpers,
} from 'utils'

class ItemMansory extends Component {
  state = {
    hover: false,
  }

  renderSourceImage = (hover, item) => {
    if (hover) {
      return item.images.fixed_height_downsampled.webp
    }
    return item.images['480w_still'].url
  }

  handleHover = value => () => {
    this.setState({
      hover: value,
    })
  }

  render() {
    const {
      placeholder,
      item,
      color,
    } = this.props

    const {
      hover,
    } = this.state

    return (
      <div
        className={Styles.mansoryItem}
        onMouseEnter={this.handleHover(true)}
        onMouseLeave={this.handleHover(false)}
      >
        <div className={Styles.positionRelative}>
          <span className={Styles.triangle(color)} />
          <span className={Styles.play(color)} />
          <div className={Helpers.mergeCss(Styles.mansoryItemChild, Styles.alignItemsEnd)}>
            <h5>
              {item.title}
            </h5>
          </div>
          <img
            alt=""
            className={Helpers.mergeCss(
              Styles.imgFluid,
              Styles.imgCover,
              Styles.backgroundColor(placeholder),
            )}
            src={this.renderSourceImage(hover, item)}
          />
        </div>
      </div>
    )
  }
}

ItemMansory.propTypes = {
  item: PropTypes.object,
  placeholder: PropTypes.object,
  color: PropTypes.string,
}

ItemMansory.defaultProps = {
  item: {},
  placeholder: {},
  color: '',
}

export default ItemMansory
