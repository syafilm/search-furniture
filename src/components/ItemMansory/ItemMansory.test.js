import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ItemMansory from './ItemMansory'

describe('Component ItemMansory', () => {
  const color = '#000'
  const item = {
    id: 'DTXugNB5Jt6Ra',
    title: 'just tell me what you want and ill be that for you ryan gosling GIF',
    images: {
      fixed_height_downsampled: {
        webp: 'https://media1.giphy.com/media/pjnfNhaFmkhxu/200_d.webp?cid=193ba05e5d259554634770392ebcb350&rid=200_d.webp',
      },
      '480w_still': {
        url: 'https://media3.giphy.com/media/pjnfNhaFmkhxu/480w_s.jpg?cid=193ba05e5d259554634770392ebcb350&rid=480w_s.jpg',
      },
    },
  }
  const data = {
    color: '#000',
    height: '186px',
  }
  it('Can render ItemMansory smoothly', () => {
    const component = shallow(
      <ItemMansory
        key={item.slug}
        placeholder={data}
        item={item}
        color={color}
      />,
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
    expect(component.find('div')).toBeDefined()
  })
})
