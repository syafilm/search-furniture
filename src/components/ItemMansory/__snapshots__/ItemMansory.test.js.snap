// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ItemMansory Can render ItemMansory smoothly 1`] = `
<div
  className="css-1ynbocu"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="css-cssveg"
  >
    <span
      className="css-n11tlf"
    />
    <span
      className="css-1te34s2"
    />
    <div
      className="css-16r1lim"
    >
      <h5>
        just tell me what you want and ill be that for you ryan gosling GIF
      </h5>
    </div>
    <img
      alt=""
      className="css-1utsrk3"
      src="https://media3.giphy.com/media/pjnfNhaFmkhxu/480w_s.jpg?cid=193ba05e5d259554634770392ebcb350&rid=480w_s.jpg"
    />
  </div>
</div>
`;
